fe/lib/components/SalaryDetails/SalaryDetails.tsx

```diff
@@ -87,6 +122,23 @@ export const SalaryDetails = ({
     min: validateMinAmount(min, errors),
   };
 
+  if (
+    payTypesLoading ||
+    currenciesLoading ||
+    payTypesData === undefined ||
+    currenciesData === undefined
+  ) {
+    return <Loader />;
```

Comment:
Nit: maybe not for this PR, but do we want a nicer loading state that results in a smaller layout shift when the data is finally loaded?

I think "Pay details" as a heading for example would be useful to keep around even while loading.